File: backend/src/DeliveryApp.Domain/Entities/Order.cs
using DeliveryApp.Domain.Common;
using DeliveryApp.Domain.Enums;

namespace DeliveryApp.Domain.Entities;

public sealed class Order : AuditableEntity
{
    public string PublicOrderNumber { get; set; } = string.Empty;

    public Guid PartnerBusinessId { get; set; }
    public PartnerBusiness PartnerBusiness { get; set; } = null!;

    public Guid CustomerId { get; set; }
    public Customer Customer { get; set; } = null!;

    public Guid PickupAddressId { get; set; }
    public Address PickupAddress { get; set; } = null!;

    public Guid DeliveryAddressId { get; set; }
    public Address DeliveryAddress { get; set; } = null!;

    public OrderStatus Status { get; set; } = OrderStatus.Available;

    public DateTime? PickupWindowStartUtc { get; set; }
    public DateTime? PickupWindowEndUtc { get; set; }
    public DateTime? DeliveryWindowStartUtc { get; set; }
    public DateTime? DeliveryWindowEndUtc { get; set; }

    public string? Notes { get; set; }
    public string? SpecialInstructions { get; set; }

    public Guid CreatedByUserId { get; set; }
    public User CreatedByUser { get; set; } = null!;

    public DateTime? CancelledAtUtc { get; set; }
    public DateTime? DeliveredAtUtc { get; set; }

    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
    public ICollection<OrderAssignment> Assignments { get; set; } = new List<OrderAssignment>();
    public ICollection<OrderStatusHistory> StatusHistory { get; set; } = new List<OrderStatusHistory>();
}

File: frontend/delivery-driver-web/src/entities/order.ts
import type { AddressDto } from './address';
import type { CustomerSummaryDto } from './customer';
import type { PartnerBusinessSummaryDto } from './partnerBusiness';

export type OrderStatus = 'Available' | 'Accepted' | 'PickedUp' | 'Delivered' | 'Cancelled';
export type OrderAction = 'Accept' | 'MarkPickedUp' | 'MarkDelivered' | 'Cancel';

export interface OrderListItemDto {
  id: string;
  publicOrderNumber: string;
  status: OrderStatus;
  partnerBusinessName: string;
  customerName: string;
  pickupAddressSummary: string;
  deliveryAddressSummary: string;
  pickupWindowStartUtc?: string | null;
  pickupWindowEndUtc?: string | null;
  deliveryWindowStartUtc?: string | null;
  deliveryWindowEndUtc?: string | null;
  itemSummary: string;
  hasSpecialInstructions: boolean;
}

export interface OrderItemDto {
  id: string;
  itemType: string;
  description: string;
  quantity: number;
  estimatedWeightKg?: number | null;
  notes?: string | null;
}

export interface OrderAssignmentDto {
  driverId: string;
  driverName: string;
  acceptedAtUtc: string;
  pickedUpAtUtc?: string | null;
  deliveredAtUtc?: string | null;
}

export interface OrderStatusHistoryDto {
  id: string;
  fromStatus?: string | null;
  toStatus: string;
  changedBy: string;
  changedAtUtc: string;
  reason?: string | null;
  notes?: string | null;
}

export interface OrderDetailsDto {
  id: string;
  publicOrderNumber: string;
  status: OrderStatus;
  partnerBusiness: PartnerBusinessSummaryDto;
  customer: CustomerSummaryDto;
  pickupAddress: AddressDto;
  deliveryAddress: AddressDto;
  items: OrderItemDto[];
  assignment?: OrderAssignmentDto | null;
  statusHistory: OrderStatusHistoryDto[];
  availableActions: OrderAction[];
  pickupWindowStartUtc?: string | null;
  pickupWindowEndUtc?: string | null;
  deliveryWindowStartUtc?: string | null;
  deliveryWindowEndUtc?: string | null;
  notes?: string | null;
  specialInstructions?: string | null;
  createdAtUtc: string;
  updatedAtUtc?: string | null;
}

export interface OrderQueryParameters {
  pageNumber?: number;
  pageSize?: number;
  search?: string;
  sortBy?: string;
  sortDirection?: 'asc' | 'desc';
}
